### ISSUE TITLE
Retroarch runs games too fast when CRTSwithRes 120Hz is enabled

### ISSUE THREAD

**Issue by MiltosKoutsokeras** [libretro/RetroArch#13630]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

## Description

Every game tested in various cores (blastem, bsnes, mame, mednafen_pce and others) is running extremely fast when CRT Switch Res is enabled for VGA monitors (horizontal 31KHz or more) in 120Hz vertical refresh mode (for proper scanlines). This behavior was not present in older builds, latest confirmed to work was 1.9.4.

I always build my own binaries, this is the commands used this time:

```Bash
./configure --prefix="${PREFIX_RETROARCH}" --disable-ffmpeg --disable-networkgamepad --disable-netplaydiscovery --enable-builtinzlib --enable-sse --disable-qt --disable-discord --disable-accessibility --disable-translate --disable-shaderpipeline

make

make install
```

### Expected behavior

Enabling CRT Switch Res 120Hz when using a VGA Monitor, Retroarch produces a video mode that has the exact line count as the original hardware, a large number of pixels in the horizontal axis (super resolution, 1920 or more) and the vertical refresh rate is usually double the content's original vertical refresh rate (60Hz games become 120Hz) in order to match the high pixel clock of the display and graphics hardware while maintaining real scanline gaps. It is glorious :)

### Actual behavior

The video mode switches to ~120Hz and a proper line count with visible scanline gaps, but the games all run extremely fast, as if there is no limit to their internal display rate. I am not sure if the follow the ~120Hz vertical refresh or they run as fast as hardware can.

### Steps to reproduce the bug

1. Launch Retroarch
2. Switch to CRTSwitchRes 120Hz
3. Launch any game in any core

### Bisect Results

This started happening in builds after 1.9.4 but I missed few of them so I cannot pinpoint which version introduced the problem.

### Version/Commit
You can find this information under Information/System Information

- RetroArch: [1.9.14/5bacf1e9be82173ed6d547c83bc7d572f085da35] and earlier but newer than v1.9.4.

### Environment information

- OS: Debian GNU/Linux 11
- Compiler: gcc (Debian 10.2.1-6) 10.2.1 20210110

**Comment by MiltosKoutsokeras** [comment]
Same problem is happening on latest version v1.10.0, commit `5bacf1e9be82173ed6d547c83bc7d572f085da35`

**Comment by briaguya-ai** [comment]
looks like there might be something in here, not sure what though

https://github.com/libretro/RetroArch/blob/c092fa33e13251fd8cf307aa3fda3b236bc99d41/gfx/video_driver.c#L3003
https://github.com/libretro/RetroArch/blob/c092fa33e13251fd8cf307aa3fda3b236bc99d41/gfx/video_driver.c#L3011
https://github.com/libretro/RetroArch/blob/c092fa33e13251fd8cf307aa3fda3b236bc99d41/gfx/video_driver.c#L3018-L3030

**Comment by RobLoach** [comment]
Audio sync enabled?

**Comment by MiltosKoutsokeras** [comment]
The issue is resolved in commit `9a3b8f4`.
